src/Remapping/distributed_remapping.jl

```diff
@@ -0,0 +1,297 @@
+
+    horz_mesh = horizontal_topology.mesh
+
+    interpolation_coeffs = map(local_target_hcoords) do hcoord
+        helem = Meshes.containing_element(horz_mesh, hcoord)
+        quad = Spaces.quadrature_style(space)
+        quad_points, _ = Spaces.Quadratures.quadrature_points(FT, quad)
+        if is_1d
+            ξ1, = Meshes.reference_coordinates(horz_mesh, helem, hcoord)
+            WI1 = Spaces.Quadratures.interpolation_matrix(
+                SVector(ξ1),
+                quad_points,
+            )
+            return (WI1,)
+        else
+            ξ1, ξ2 = Meshes.reference_coordinates(horz_mesh, helem, hcoord)
+            WI1 = Spaces.Quadratures.interpolation_matrix(
+                SVector(ξ1),
+                quad_points,
+            )
+            WI2 = Spaces.Quadratures.interpolation_matrix(
+                SVector(ξ2),
+                quad_points,
+            )
+            return (WI1, WI2)
+        end
+    end
+
+    # For 2D meshes, we have a notion of local and global indices. This is not the case for
+    # 1D meshes, which are much simpler. For 1D meshes, the "index" is the same as the
+    # element number, for 2D ones, we have to do some work.
+    if is_1d
+        local_indices = map(local_target_hcoords) do hcoord
+            return Meshes.containing_element(horz_mesh, hcoord)
+        end
+    else
+        # We need to obtain the local index from the global, so we prepare a lookup table
+        global_elem_lidx = Dict{Int, Int}() # inverse of local_elem_gidx: lidx = global_elem_lidx[gidx]
+        for (lidx, gidx) in enumerate(horizontal_topology.local_elem_gidx)
+            global_elem_lidx[gidx] = lidx
+        end
+
+        local_indices = map(local_target_hcoords) do hcoord
+            helem = Meshes.containing_element(horz_mesh, hcoord)
+            return global_elem_lidx[horizontal_topology.orderindex[helem]]
+        end
+    end
+
+    # We represent interpolation onto an horizontal slab as an empty list of zcoords
+    isnothing(target_zcoords) && (target_zcoords = [])
+
+    return Remapper(
+        comms_ctx,
+        local_target_hcoords,
+        target_zcoords,
+        local_target_hcoords_bitmask,
+        interpolation_coeffs,
+        local_indices,
+        space,
+    )
+end
+
+
+"""
+   interpolate(remapper, field)
+
+Interpolate the given `field` as prescribed by `remapper`.
+
+Example
+========
+
+Given `field1`,`field2`, two `Field` defined on a cubed sphere.
+
+```julia
+longpts = range(-180.0, 180.0, 21)
+latpts = range(-80.0, 80.0, 21)
+zpts = range(0.0, 1000.0, 21)
+
+hcoords = [Geometry.LatLongPoint(lat, long) for long in longpts, lat in latpts]
+zcoords = [Geometry.ZPoint(z) for z in zpts]
+
+space = axes(field1)
+
+remapper = Remapper(hcoords, zcoords, space)
+
+int1 = interpolate(remapper, field1)
+int2 = interpolate(remapper, field2)
+
+```
+"""
+function interpolate(remapper::Remapper, field::T) where {T <: Fields.Field}
+
+    axes(field) == remapper.space ||
```

Comment:
```suggestion
    axes(field) === remapper.space ||
```
I think we want to compare the pointer here

Reply:
I think that's what's already happening since `Space`s don't define `==`. 

From a philosophical point of view, I would like if two spaces are interchangeable, not if they are identical. The code works if you have to distinct but equivalent spaces (even if this concept is currently blurred because `Space`s are immutable.

Reply:
Ah, right. And I agree with your philosophical point.